### ISSUE TITLE
process-to-connection mapping on OSX

### ISSUE THREAD

**Issue by raboof** [raboof/nethogs#61]
We establish the process-to-connection (or, more accurately, process-to-inode) mapping based on `/proc/<pid>/fd` on Linux.

Does OSX have something similar?

**Comment by mattmo** [comment]
lsof should be able to provide this and i believe it comes preinstalled on osx. here's example output from my mac:

`sudo lsof | grep 162.222.42.179

CrashPlan   57            root  112u     IPv6 0x3ff9715c9699fad3        0t0       TCP 172.19.131.151:64352->162.222.42.179:https (ESTABLISHED)`

**Comment by mattmo** [comment]
CrashPlan is the process name and 57 is the PID

**Comment by zaaack** [comment]
Is any update here? It is an important feature

**Comment by raboof** [comment]
@zaaack I agree this would be a useful feature, but at the moment I don't have access to a mac myself and am not planning to implement it.

A community contribution adding this feature would be most welcome though! (We shouldn't have to invoke lsof to get this information, but perhaps use the same underlying mechanism that's also used by lsof)

**Comment by rkrp** [comment]
@raboof I recently got a Mac and I very much would love to work on this one. Although I have decent experience working with C++, I am completely new to the codebase. 

I am currently looking for ways to get the same info from `lsof` without invoking it as you previously suggested. 

I would really appreciate some pointers on where to get started 😄

**Comment by rkrp** [comment]
I am currently reading this [article](http://blog.palominolabs.com/2012/06/19/getting-the-files-being-used-by-a-process-on-mac-os-x/) for getting info about open file descriptors in macOS and it looks very promising. Also reading the documentation present in `lsof` [source code](https://github.com/practicalswift/osx/tree/master/src/lsof/lsof) hoping that it might lead somewhere.

**Comment by scottcc** [comment]
+1 for this.. though based on the history above I'm not as optimistic as I'd like to be.
